region/crateria/west.json

```diff
@@ -1792,62 +1809,71 @@
                 {
                   "name": "Shinespark Entry",
                   "notable": false,
-                  "requires": [],
+                  "requires": [
+                    {"canComeInCharged": {
+                      "fromNode": 1,
+                      "framesRemaining": 0,
+                      "shinesparkFrames": 90,
+                      "excessShinesparkFrames": 17
+                    }}
+                  ],
                   "obstacles": [
+                    {
+                      "id": "A",
+                      "requires": []
+                    },
                     {
                       "id": "B",
-                      "requires": [
-                        {"canComeInCharged": {
-                          "fromNode": 1,
-                          "framesRemaining": 0,
-                          "shinesparkFrames": 90,
-                          "excessShinesparkFrames": 17
-                         }}
-                      ],
-                      "additionalObstacles": ["A"],
-                      "note": "Enter through the top of the door to reach all the wall to the shot block wall."
+                      "requires": []
                     }
-                  ]
+                  ],
+                  "note": "Enter through the top of the door to reach all the way to the shot block wall."
```

Comment:
Going through the top of the doorway is likely a midair, although, shinesparking from the bottom of the door will get most of the way through, only breaking the bottom block near the end, but there is no real reason to go through this room without morph anyway. (Other than the off chance you're only entering to view the item, but I'd say that isnt really a priority yet)